src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -245,6 +248,51 @@ def export_config(cmd,
         __write_kv_to_app_service(cmd, key_values=src_kvs, appservice_account=appservice_account)
 
 
+def export_secret(cmd,
+                  name=None,
+                  connection_string=None,
+                  label=None,
+                  key=None,
+                  prefix="",
+                  path=None,
+                  format_=None):
+    # fetch key values from user's configstore
+    src_kvs = __read_kv_from_config_store(
+        cmd, name=name, connection_string=connection_string, key=key, label=label, prefix_to_remove=prefix)
+
+    kv_references = [keyvaule for keyvaule in src_kvs if keyvaule.content_type == KeyVaultConstants.KEYVAULT_CONTENT_TYPE]
+    keyvault_client = __get_keyvault_client(cmd.cli_ctx)
+    secrets = {}
+
+    from azure.keyvault.key_vault_id import SecretId
+    for kv in kv_references:
+        try:
+            secret_id = json.loads(kv.value)["uri"]
+            kv_identifier = SecretId(uri=secret_id)
+        except (TypeError, ValueError) as error:
+            logger.error("Invalid key vault reference for key:%s value:%s. Error:%s", kv.key, kv.value, error)
+            continue
+
+        try:
+            secret = keyvault_client.get_secret(vault_base_url=kv_identifier.vault,
+                                                secret_name=kv_identifier.name,
+                                                secret_version=kv_identifier.version)
+            secrets[kv.key] = secret.value
+        except Exception as exception:   # pylint: disable=broad-except
+            logger.error("Cannot fetch secret: %s. Exception: %s", secret_id, str(exception))
+
+    if path:
+        with open(path, 'w') as fp:
```

Comment:
>with open(path, 'w') as [](start = 8, length = 23)

Do we want to add encoding='utf-8' here?

Reply:
Yeah, we should use utf8 as default encoding. However, from keyvault perspective, each secret can have its own encoding. For instance ```az keyvault secret download --encoding```. Since for our command, we kind of batch export secrets into one file, there is no way to customize encoding for each secret.